File: Lab 2/lab2.py
print("-- task 1 --")
number = int(input("enter a number greater than one:"))
steps = 0

while number>1:
    steps+=1
    print(number, end=" -> ")
    if number%2==0:
        number=number//2
    else:
        number=number*3+1

print(1)
print("total steps:", steps)
print(" ")


print("-- task 2 --")
x = int(input("enter a number between 10 and 100:"))
while 10>x or x>100:
    x = int(input("invalid input.  enter a number between 10 and 100:"))
count=0
fizz=0
buzz=0
fizzbuzz=0
while count<x:
    count+=1
    if count%7==0:
        continue
    if count%3==0 and count%5==0:
        print("FizzBuzz")
        fizzbuzz+=1
    elif count%3==0:
        print("Fizz")
        fizz+=1
    elif count%5==0:
        print("Buzz")
        buzz+=1
    else:
        print(count)
    
print("- summary -")
print("Fizz count: ", fizz)
print("Buzz count: ", buzz)
print("FizzBuzz count: ", fizzbuzz)
print(" ")

print("-- task 3 --")
pattern = int(input("Please enter a number between 3 and 9: "))
if 3 <= pattern <= 9:
    for i in range(1, 2 * pattern):
        level = pattern - abs(pattern - i)
        print("*" * level)

File: Lab 2/lab2.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {
    cout << "-- task 1 --" << endl;
    int number;
    int steps = 0;
    cout << "enter a positive integer greater than 1: ";
    cin >> number;

    while (number > 1) {
        cout << number << " -> ";
        if (number % 2 == 0) {
            number = number / 2;
        } else {
            number = (number * 3) + 1;
        }
        steps++;
    }
    cout << 1 << endl;
    cout << "total steps: " << steps << endl;


    cout << "\n-- task 2 --" << endl;
    int n;
    while (true) {
        cout << "enter a number between 10 and 100: ";
        cin >> n;
        if (n >= 10 && n <= 100) {
            break;
        }
        cout << "invalid input. ";
    }

    int fizz = 0, buzz = 0, fizzbuzz = 0;
    for (int i = 1; i <= n; i++) {
        if (i % 7 == 0) {
            continue;
        }
        if (i % 3 == 0 && i % 5 == 0) {
            cout << "FizzBuzz" << endl;
            fizzbuzz++;
        } else if (i % 3 == 0) {
            cout << "Fizz" << endl;
            fizz++;
        } else if (i % 5 == 0) {
            cout << "Buzz" << endl;
            buzz++;
        } else {
            cout << i << endl;
        }
    }
     cout << " " << endl;
    cout << "- summary -" << endl;
    cout << "Fizz count: " << fizz << endl;
    cout << "Buzz count: " << buzz << endl;
    cout << "FizzBuzz count: " << fizzbuzz << endl;


    cout << "\n-- task 3 --" << endl;
    int x;
    cout << "Enter a number (3-9): ";
    cin >> x;
    for (int i = 1; i <= 2 * x - 1; i++) {
        int stars = x - abs(x - i); 

        for (int j = 0; j < stars; j++) {
            cout << "*";
        }
        cout << endl; 
    }

    return 0;
}